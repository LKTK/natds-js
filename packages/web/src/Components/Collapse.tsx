import React, { FunctionComponent } from 'react';
import MaterialCollapse, {
  CollapseProps as ICollapseProps,
} from '@material-ui/core/Collapse';
export { CollapseProps as ICollapseProps } from '@material-ui/core/Collapse';

const Collapse: FunctionComponent<ICollapseProps> = (props: ICollapseProps) => (
  <MaterialCollapse {...props} />
);

export default Collapse;
