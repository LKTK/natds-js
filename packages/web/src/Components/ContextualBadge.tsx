import * as React from 'react';
import styled from 'styled-components';
import { withTheme, createMuiTheme } from '@material-ui/core/styles';
import isEqual from 'lodash/isEqual';

import { IThemeWeb, themes } from '../Themes';

const defaultValues = {
  lineHeight: 16
};

type IContextualBadgeColors = 'primary' | 'secondary' | 'info' | 'error' | 'warning' | 'success' | 'light' | 'dark';

type IContextualBadgeTypes = 'standard';

export interface IContextualBadgeProps {
  [propName: string]: any;
  /**
   * The input id property
   */
  id: string;
  /**
   * @optional
   */
  theme?: IThemeWeb | unknown;
  /**
   * @optional
   * Color of the ContextualBadge. Defaults to 'primary'
   */
  color?: IContextualBadgeColors;
  /**
   * @optional
   * Type of the ContextualBadge. Defaults to 'standard'
   */
  type?: IContextualBadgeTypes;
  /**
   * Classes to be used on the ContextualBadge
   */
  classes?: string;
  /**
   * Style to be used on the ContextualBadge
   */
  style?: React.CSSProperties;
  /**
   * Elements to be placed inside the ContextualBadge
   */
  children?: React.ReactNode;
}

export const ContextualBadge: React.FunctionComponent<IContextualBadgeProps> = (props: IContextualBadgeProps) => {
  const {
    theme: providerTheme,
    color = 'primary',
    type = 'standard',
    classes,
    style,
    children,
    ...rest
  } = props;

  const theme: any = React.useMemo(() => {
    const parsedProviderTheme = JSON.parse(JSON.stringify(providerTheme));
    const parsedDefaulttheme = JSON.parse(JSON.stringify(createMuiTheme({})));
    return isEqual(parsedProviderTheme, parsedDefaulttheme)
      ? createMuiTheme(themes.natura.light)
      : providerTheme;
  },
    [providerTheme]);

  const colorMap = React.useMemo(() => {
    const { primary, secondary, complementary, error, success, background, text } = theme.palette;
    return {
      primary: {
        container: primary.main,
        text: primary.contrastText
      },
      secondary: {
        container: secondary.main,
        text: secondary.contrastText
      },
      info: {
        container: complementary.link,
        text: complementary.linkContrastText
      },
      error: {
        container: error.main,
        text: error.contrastText
      },
      warning: {
        container: complementary.warning,
        text: complementary.warningContrastText
      },
      success: {
        container: success.main,
        text: success.contrastText
      },
      light: {
        container: background.paper,
        text: background.paperContrastText
      },
      dark: {
        container: text.disabled,
        text: background.defaultContrastText
      }
    };
  }, [theme]);

  const fontMap = React.useMemo(() => {
    const { typography } = theme;
    return {
      fontFamily: typography.fontFamily,
      fontSize: typography.caption.fontSize,
      fontWeight: typography.caption.fontWeight,
      lineHeight: defaultValues.lineHeight
    };
  }, [theme]);

  const constraints = React.useMemo(() => {
    const { spacing, shape } = theme;
    return {
      padding: `1px ${spacing()}px`,
      borderRadius: shape.badgeBorderRadius
    };
  }, [theme]);

  return (
    <Container
      backgroundColor={colorMap[color].container}
      textColor={colorMap[color].text}
      fontFamily={fontMap.fontFamily}
      fontSize={fontMap.fontSize}
      fontWeight={fontMap.fontWeight}
      lineHeight={fontMap.lineHeight}
      padding={constraints.padding}
      borderRadius={constraints.borderRadius}
      style={style}
      className={classes}
      {...rest}
    >
      {props.children}
    </Container>
  );
};

interface IContainerProps {
  backgroundColor: string;
  textColor: string;
  fontFamily: string;
  fontSize: string | number;
  fontWeight: string | number;
  lineHeight: number;
  padding: string;
  borderRadius: number;
}

const Container = styled.div<IContainerProps> `
  background-color: ${props => props.backgroundColor};
  color: ${props => props.textColor};
  font-family: ${props => props.fontFamily};
  font-size: ${props => props.fontSize};
  font-weight: ${props => props.fontWeight};
  line-height: ${props => props.lineHeight}px;
  padding: ${props => props.padding};
  border-radius: ${props => props.borderRadius}px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export default withTheme(ContextualBadge);
