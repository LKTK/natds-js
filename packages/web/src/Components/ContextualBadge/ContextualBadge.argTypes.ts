import { ArgType } from "@storybook/addons";
import { IContextualBadgeProps } from "./ContextualBadge.props";
import { colors } from "./__fixtures__/colors";
import { types } from "./__fixtures__/types";

export const argTypes : Record<keyof IContextualBadgeProps, ArgType> = {
  classes: {
    control: { type: "text" },
    description: "The classes to use on the container 'div'",
    table: {
      type: { summary: "string" },
    },
    type: { name: "string", required: false },
  },
  children: {
    description: "The children that will be placed inside the badge",
    table: {
      type: { summary: "React.ReactNode" },
    },
  },
  color: {
    control: {
      options: colors,
      type: "select",
    },
    description: "Color of the ContextualBadge to use from the theme.",
    table: {
      defaultValue: { summary: "\"primary\"" },
      type: { summary: "'primary' \\| 'secondary' \\| 'info' \\| 'error' \\| 'warning' \\| 'success' \\| 'light' \\| 'dark'" },
    },
  },
  id: {
    control: { type: "text" },
    description: "The input id property",
    table: {
      type: { summary: "string" },
    },
    type: { name: "string", required: true },
  },
  style: {
    control: { type: "object" },
    description: "Inline style for the container 'div'",
    table: {
      type: { summary: "React.CSSProperties" },
    },
    type: { name: "object", required: false },
  },
  type: {
    control: {
      options: types,
      type: "inline-radio",
    },
    description: "The type of the badge",
    table: {
      defaultValue: { summary: "\"standard\"" },
    },
  },
};

export default argTypes;
