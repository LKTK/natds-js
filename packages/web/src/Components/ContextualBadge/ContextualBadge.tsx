// eslint-disable-next-line no-use-before-define
import * as React from "react";
import clsx from "clsx";
import { IContextualBadgeProps } from "./ContextualBadge.props";
import useStyles from "./ContextualBadge.styles";

export { IContextualBadgeProps } from "./ContextualBadge.props";

/**
 * ## Importing
 *
 * ```
 * import { ContextualBadge } from '@naturacosmeticos/natds-web';
 * ```
 *
 * @todo rename component to Tag
 */
export const ContextualBadge = React.forwardRef<HTMLDivElement, IContextualBadgeProps>(
  (props: IContextualBadgeProps, ref) => {
    const {
      children,
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      classes: legacyClasses,
      className,
      component: Component = "span",
      color = "primary",
      style,
      type = "standard",
      ...otherProps
    } = props;

    const classes = useStyles();

    return (
      <Component
        className={clsx(classes.root, classes[color], legacyClasses, className)}
        data-type={type}
        ref={ref}
        style={style}
        {...otherProps}
      >
        {children}
      </Component>
    );
  });

ContextualBadge.displayName = "ContextualBadge";

export default ContextualBadge;
