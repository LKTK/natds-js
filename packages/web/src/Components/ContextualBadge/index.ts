import ContextualBadge from "./ContextualBadge";

export { IContextualBadgeProps } from "./ContextualBadge.props";
export { ContextualBadge };
export default ContextualBadge;
