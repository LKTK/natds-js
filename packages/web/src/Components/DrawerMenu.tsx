import React, { FunctionComponent } from 'react';
import { withTheme, makeStyles } from '@material-ui/core';
import styled from 'styled-components';
import { IThemeWeb } from 'Themes';
import { getDefaultTheme } from './shared';

import ContextualBadge from './ContextualBadge';
import Collapse from './Collapse';
import Icon from './Icon';
import List from './List';
import ListItem from './ListItem';
import ListSubheader from './ListSubheader';
import ListItemIcon from './ListItemIcon';
import ListItemText from './ListItemText';
import Typography from './Typography';

import { iconNames } from '@naturacosmeticos/natds-icons';

const ICON_ARROW_UP = 'outlined-navigation-arrowtop';
const ICON_ARROW_DOWN = 'outlined-navigation-arrowbottom';

export interface IDrawerMenuProps {
  children?: React.ReactNode;
  component?: React.ElementType;
  list?: [IDrawerMenuSectionProps];
  theme: IThemeWeb | unknown;
}

export interface IDrawerMenuItemProps {
  name?: string;
  section?: string;
  icon?: keyof typeof iconNames;
  onSelect?: (event: React.SyntheticEvent, name?: string) => void;
  badge?: string;
  selected?: boolean;
  theme?: IThemeWeb | unknown;
}

export interface IDrawerMenuSectionProps extends IDrawerMenuItemProps {
  onToggle?: (event: React.SyntheticEvent, state: boolean, name?: string) => void;
  opened?: boolean;
  list?: [IDrawerMenuSectionProps];
}

export const DrawerMenu: FunctionComponent<IDrawerMenuProps> = (props: IDrawerMenuProps) => {
  const { children, list, theme: providerTheme, component, ...rest } = props;
  const theme: any = React.useMemo(() => getDefaultTheme(providerTheme), [providerTheme]);
  const StyledList = React.useMemo(() => makeStyles({
    padding: {
      paddingTop: 0,
      paddingBottom: 0,
      marginTop: theme.sizes.tiny,
      marginBottom: theme.sizes.tiny
    }
  }), [theme]);

  const content = children ? children : (
    <List
      children={list && list.map(BuildDrawerMenuItems(theme))}
      classes={StyledList()}
      dense
    />
  );

  return (
    <DrawerMenuComponent {...rest} as={component}>
      {content}
    </DrawerMenuComponent>
  );
};

export const DrawerMenuSection: FunctionComponent<IDrawerMenuSectionProps> = (props: IDrawerMenuSectionProps) => {
  const { onToggle, icon, name, list, theme: providerTheme } = props;
  const theme: any = React.useMemo(() => getDefaultTheme(providerTheme), [providerTheme]);
  const [opened, toggleSubmenu] = React.useState(props.opened);

  const StyledSubList = React.useMemo(() => makeStyles({
    padding: {
      borderLeft: `${theme.palette.action.hover} 1px solid`,
      padding: `0 0 0 ${theme.sizes.tiny}px`,
      margin: `0 0 0 ${theme.sizes.semix}px`
    }
  }), [theme]);

  const StyledListSectionItem = React.useMemo(() => makeStyles({
    root: {
      borderRadius: theme.shape.borderRadius,
      padding: theme.sizes.tiny,
      marginBottom: theme.sizes.micro,
      cursor: 'pointer',
      '&:hover': {
        backgroundColor: theme.palette.action.hover
      },
      '&$selected': {
        backgroundColor: `${theme.palette.action.hover}`,
        '&:hover': {
          backgroundColor: theme.palette.action.hover
        }
      }
    },
    selected: {}
  }), [theme]);

  const StyledListItemIcon = React.useMemo(() => makeStyles({
    root: {
      minWidth: 'auto',
      marginRight: theme.sizes.standard,
      marginTop: 0
    }
  }), [theme]);

  const classes = StyledSubList();
  const classesItem = StyledListSectionItem();
  const classesIcon = StyledListItemIcon();
  const listIconName = opened ? ICON_ARROW_UP : ICON_ARROW_DOWN;

  const handleClick = (event: React.SyntheticEvent) => {
    toggleSubmenu(!opened);

    onToggle && onToggle(event, !opened, name);
  };

  return (
    <>
      <ListItem onClick={handleClick} classes={classesItem} selected={opened} component="li" button>
        {icon && <ListItemIcon classes={classesIcon} children={<Icon name={icon} size="tiny" />} />}
        <ListItemText children={getMenuItemText(props)} />
        <Icon name={listIconName} size="tiny" />
      </ListItem>
      <Collapse in={opened} timeout="auto" unmountOnExit component="li">
        {list && <List
          children={list.map(BuildDrawerMenuItems(theme))}
          classes={classes}
          dense
        />}
      </Collapse>
    </>
  );
};

export const DrawerMenuItem: FunctionComponent<IDrawerMenuItemProps> = (props: IDrawerMenuItemProps) => {
  const { onSelect, name, icon, selected, section, theme: providerTheme } = props;
  const theme: any = React.useMemo(() => getDefaultTheme(providerTheme), [providerTheme]);

  const StyledListItem = React.useMemo(() => makeStyles({
    root: {
      borderRadius: theme.shape.borderRadius,
      padding: theme.sizes.tiny,
      marginBottom: theme.sizes.micro,
      cursor: 'pointer',
      '&:hover': {
        backgroundColor: theme.palette.action.hover
      }
    }
  }), [theme]);

  const StyledListSubheader = React.useMemo(() => makeStyles({
    root: {
      backgroundColor: `${theme.palette.background.paper}`,
      borderTop: `${theme.palette.action.hover} 1px solid`,
      margin: `0 -${theme.sizes.tiny}px`
    }
  }), [theme]);

  const StyledListItemIcon = React.useMemo(() => makeStyles({
    root: {
      minWidth: 'auto',
      marginRight: theme.sizes.standard,
      marginTop: 0
    }
  }), [theme]);

  if (section) {
    const classesSubheader = StyledListSubheader(theme);

    return (
      <ListSubheader classes={classesSubheader} children={
        <Typography variant="caption" component="span" children={section} />
      } />
    );
  }

  const classes = StyledListItem();
  const classesIcon = StyledListItemIcon();
  const handleClick = (event: React.SyntheticEvent) => onSelect && onSelect(event, name);
  const onClick = onSelect ? { onClick: handleClick } : {};

  return (
    <ListItem {...onClick} classes={classes} selected={selected} component="li" button>
      {icon && <ListItemIcon classes={classesIcon} children={<Icon name={icon} size="tiny" />} />}
      <ListItemText children={getMenuItemText(props)} />
    </ListItem>
  );
};

const BuildDrawerMenuItems = (theme: IThemeWeb | unknown) => (
  (props: IDrawerMenuSectionProps, key: number) => (
    props.list
      ? <DrawerMenuSection theme={theme} key={key} {...props} />
      : <DrawerMenuItem theme={theme} key={key} {...props} />
  )
);

const getMenuItemText = (props: IDrawerMenuSectionProps) => {
  const { badge, name, theme } = props;

  if (!badge) return name;

  return (
    <>
      <Typography variant="body2" component="span" children={name} />
      <StyledBadge as={ContextualBadge} theme={theme} color="error" children={badge} />
    </>
  );
};

const DrawerMenuComponent = styled.div`
  flex: 1 1 auto;
`;

const StyledBadge = styled.div<{ theme: IThemeWeb }>`
  margin-left: ${({ theme }) => `${theme.sizes.micro}px`};
`;

export default withTheme(DrawerMenu);
