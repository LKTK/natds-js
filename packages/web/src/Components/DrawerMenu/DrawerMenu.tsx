import * as React from "react";
import { makeStyles } from "@material-ui/core";
import List from "../List";
import { IDrawerMenuProps } from "./DrawerMenu.props";
import useDefaultTheme from "../../hooks/useDefaultTheme";
import { BuildDrawerMenuItems } from "./BuildDrawerMenuItems";
import { DrawerMenuComponent } from "./DrawerMenuComponent";

/**
 * ## Importing
 *
 * ```
 * import { DrawerMenu } from '@naturacosmeticos/natds-web';
 * ```
 *
 * @todo refactor(web): refactor DrawerMenu component
 */
export const DrawerMenu = React.forwardRef<HTMLElement, IDrawerMenuProps>((
  props: IDrawerMenuProps, ref,
) => {
  const {
    children, list, component, ...rest
  } = props;

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const theme : any = useDefaultTheme();

  // @todo move to .styles.ts file
  const StyledList = React.useMemo(() => makeStyles({
      padding: {
        marginBottom: theme.sizes?.tiny,
        marginTop: theme.sizes?.tiny,
        paddingBottom: 0,
        paddingTop: 0,
      },
    }), [theme]),

    content = children || <List
      classes={StyledList()}
      dense
    >{list && list.map((drawerMenuSectionProps, index) => (
        <BuildDrawerMenuItems {...drawerMenuSectionProps} key={index} />
      ))}
    </List>;

  return (
    <DrawerMenuComponent {...rest} as={component} ref={ref}>
      {content}
    </DrawerMenuComponent>
  );
});

DrawerMenu.displayName = "DrawerMenu";

export default DrawerMenu;
