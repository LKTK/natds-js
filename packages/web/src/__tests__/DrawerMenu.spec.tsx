import React from 'react';
import { expect } from 'chai';
import renderer from 'react-test-renderer';
import { spy, assert } from 'sinon';
import { mount } from 'enzyme';

import DrawerMenu, { IDrawerMenuSectionProps, DrawerMenuItem, DrawerMenuSection } from '../Components/DrawerMenu';
import ContextualBadge from '../Components/ContextualBadge';
import Collapse from '../Components/Collapse';

const listBasic = [
  {
    name: 'Menu Item',
    icon: 'outlined-action-love',
  }
];

const listComplete = [
  {
    name: 'Menu Item',
    icon: 'outlined-action-love',
  },
  {
    name: 'Menu Item',
    icon: 'outlined-action-love',
    badge: 'Novo',
    onSelect: () => null,
  },
  {
    name: 'Menu Item',
    icon: 'outlined-action-love',
    opened: false,
    onToggle: () => null,
    badge: 'Novo',
    list: [
      {
        name: 'Menu Item',
        icon: 'outlined-action-love',
      },
      {
        name: 'Menu Item',
        icon: 'outlined-action-love',
        badge: 'Novo',
      }
    ]
  },
  {
    section: 'section'
  },
  {
    name: 'Menu Item',
    icon: 'outlined-action-love',
    badge: 'Novo'
  },
];

describe('DrawerMenu component', () => {
  test('should match snapshot - DrawerMenu list', () => {
    const component = renderer.create(<DrawerMenu list={listComplete as [IDrawerMenuSectionProps]} />).toJSON();

    expect(component).matchSnapshot('DrawerMenu list snapshot');
  });

  test('should match snapshot - DrawerMenu custom component', () => {
    const component = renderer.create(<DrawerMenu list={listComplete as [IDrawerMenuSectionProps]} component="section" />).toJSON();

    expect(component).matchSnapshot('DrawerMenu custom component snapshot');
  });

  test('should match snapshot - DrawerMenu custom children', () => {
    const children = (<div>some custom children</div>);
    const component = renderer.create(<DrawerMenu list={listComplete as [IDrawerMenuSectionProps]} children={children} />).toJSON();

    expect(component).matchSnapshot('DrawerMenu custom children snapshot');
  });

  test('should render custom children', () => {
    const children = (<div className="custom">some custom children</div>);
    const component = mount(<DrawerMenu list={listBasic as [IDrawerMenuSectionProps]} children={children} />);

    expect(component).to.have.descendants('.custom');
    expect(component).to.not.have.descendants(DrawerMenuItem);
    expect(component).to.not.have.descendants(DrawerMenuSection);
  });

  test('should render list item', () => {
    const component = mount(<DrawerMenu list={listBasic as [IDrawerMenuSectionProps]} />);

    expect(component).to.have.descendants(DrawerMenuItem);
    expect(component).to.not.have.descendants(ContextualBadge);
    expect(component).to.not.have.descendants(DrawerMenuSection);
  });

  test('should render list item with badge', () => {
    const [listItem] = listBasic;
    const list = [{ ...listItem, badge: 'badge' }];
    const component = mount(<DrawerMenu list={list as [IDrawerMenuSectionProps]} />);

    expect(component).to.have.descendants(ContextualBadge);
    expect(component).to.not.have.descendants(DrawerMenuSection);
  });

  test('should render submenu item', () => {
    const component = mount(<DrawerMenu list={listComplete as [IDrawerMenuSectionProps]} />);

    expect(component).to.have.descendants(DrawerMenuSection);
  });

  test('should start with submenu opened', () => {
    const listItem = listComplete[2];
    const list = [{ ...listItem, opened: true }];
    const component = mount(<DrawerMenu list={list as [IDrawerMenuSectionProps]} />);

    expect(component.find(Collapse).prop('in')).to.equal(true);
  });

  test('should open submenu and call onToggle', () => {
    const onToggle = spy(listComplete[2], 'onToggle');
    const component = mount(<DrawerMenu list={listComplete as [IDrawerMenuSectionProps]} />);

    expect(component.find(Collapse).prop('in')).to.equal(false);

    component.find('li').at(2).simulate('click');

    expect(component.find(Collapse).prop('in')).to.equal(true);

    assert.calledOnce(onToggle);
  });

  test('should select a item and call onSelect', () => {
    const onSelect = spy(listComplete[1], 'onSelect');
    const component = mount(<DrawerMenu list={listComplete as [IDrawerMenuSectionProps]} />);

    component.find('li').at(1).simulate('click');

    assert.calledOnce(onSelect);
  });
});
