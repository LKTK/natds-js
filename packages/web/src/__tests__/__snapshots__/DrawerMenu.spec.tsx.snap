// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerMenu component should match snapshot - DrawerMenu custom children: DrawerMenu custom children snapshot 1`] = `
<div
  className="sc-htpNat kqTbrx"
>
  <div>
    some custom children
  </div>
</div>
`;

exports[`DrawerMenu component should match snapshot - DrawerMenu custom component: DrawerMenu custom component snapshot 1`] = `
<section
  className="sc-htpNat kqTbrx"
>
  <ul
    className="MuiList-root MuiList-dense MuiList-padding makeStyles-padding-80"
  >
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-81 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root makeStyles-root-82"
      >
        <i
          className="sc-bwzfXH iglmwN natds-icons natds-icons-outlined-action-love"
          size="tiny"
        />
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
        >
          Menu Item
        </span>
      </div>
      <span
        className="MuiTouchRipple-root"
      />
    </li>
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-83 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root makeStyles-root-84"
      >
        <i
          className="sc-bwzfXH iglmwN natds-icons natds-icons-outlined-action-love"
          size="tiny"
        />
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-bdVaJa jUDhJV sc-bxivhb gkPMhX"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize="0.75rem"
            fontWeight={400}
          >
            Novo
          </span>
        </span>
      </div>
      <span
        className="MuiTouchRipple-root"
      />
    </li>
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-86 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root makeStyles-root-88"
      >
        <i
          className="sc-bwzfXH iglmwN natds-icons natds-icons-outlined-action-love"
          size="tiny"
        />
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-bdVaJa jUDhJV sc-bxivhb gkPMhX"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize="0.75rem"
            fontWeight={400}
          >
            Novo
          </span>
        </span>
      </div>
      <i
        className="sc-bwzfXH iglmwN natds-icons natds-icons-outlined-navigation-arrowbottom"
        size="tiny"
      />
      <span
        className="MuiTouchRipple-root"
      />
    </li>
    <li
      className="MuiListSubheader-root makeStyles-root-89 MuiListSubheader-sticky MuiListSubheader-gutters"
    >
      <span
        className="MuiTypography-root MuiTypography-caption"
      >
        section
      </span>
    </li>
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-90 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root makeStyles-root-91"
      >
        <i
          className="sc-bwzfXH iglmwN natds-icons natds-icons-outlined-action-love"
          size="tiny"
        />
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-bdVaJa jUDhJV sc-bxivhb gkPMhX"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize="0.75rem"
            fontWeight={400}
          >
            Novo
          </span>
        </span>
      </div>
      <span
        className="MuiTouchRipple-root"
      />
    </li>
  </ul>
</section>
`;

exports[`DrawerMenu component should match snapshot - DrawerMenu list: DrawerMenu list snapshot 1`] = `
<div
  className="sc-htpNat kqTbrx"
>
  <ul
    className="MuiList-root MuiList-dense MuiList-padding makeStyles-padding-1"
  >
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-6 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root makeStyles-root-7"
      >
        <i
          className="sc-bwzfXH iglmwN natds-icons natds-icons-outlined-action-love"
          size="tiny"
        />
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
        >
          Menu Item
        </span>
      </div>
    </li>
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-60 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root makeStyles-root-61"
      >
        <i
          className="sc-bwzfXH iglmwN natds-icons natds-icons-outlined-action-love"
          size="tiny"
        />
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-bdVaJa jUDhJV sc-bxivhb gkPMhX"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize="0.75rem"
            fontWeight={400}
          >
            Novo
          </span>
        </span>
      </div>
    </li>
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-63 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root makeStyles-root-65"
      >
        <i
          className="sc-bwzfXH iglmwN natds-icons natds-icons-outlined-action-love"
          size="tiny"
        />
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-bdVaJa jUDhJV sc-bxivhb gkPMhX"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize="0.75rem"
            fontWeight={400}
          >
            Novo
          </span>
        </span>
      </div>
      <i
        className="sc-bwzfXH iglmwN natds-icons natds-icons-outlined-navigation-arrowbottom"
        size="tiny"
      />
    </li>
    <li
      className="MuiListSubheader-root makeStyles-root-71 MuiListSubheader-sticky MuiListSubheader-gutters"
    >
      <span
        className="MuiTypography-root MuiTypography-caption"
      >
        section
      </span>
    </li>
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-78 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root makeStyles-root-79"
      >
        <i
          className="sc-bwzfXH iglmwN natds-icons natds-icons-outlined-action-love"
          size="tiny"
        />
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-bdVaJa jUDhJV sc-bxivhb gkPMhX"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize="0.75rem"
            fontWeight={400}
          >
            Novo
          </span>
        </span>
      </div>
    </li>
  </ul>
</div>
`;
